File: fsharp/difference-of-squares/DifferenceOfSquares.fs
﻿module DifferenceOfSquares

let square x = x * x

let squareOfSum (number: int): int =
    seq {1..number} |> Seq.sum |> square

let sumOfSquares (number: int): int =
    seq {1..number} |> Seq.sumBy square

let differenceOfSquares (number: int): int =
    squareOfSum number - sumOfSquares number

File: zig/difference-of-squares/difference_of_squares.zig
pub fn squareOfSum(number: usize) usize {
    var result: usize = 0;
    for (1..number + 1) |i| {
        result += i;
    }
    return result * result;
}

pub fn sumOfSquares(number: usize) usize {
    var result: usize = 0;
    for (1..number + 1) |i| {
        result += i * i;
    }
    return result;
}

pub fn differenceOfSquares(number: usize) usize {
    return squareOfSum(number) - sumOfSquares(number);
}
